src/openMVG/sfm/sfm_data_triangulation.cpp

```diff
@@ -40,12 +40,53 @@ SfM_Data_Structure_Computation_Blind::SfM_Data_Structure_Computation_Blind
 {
 }
 
+/// Triangulate a given track
+bool track_full_triangulation
```

Comment:
`track_full_triangulation` and `track_sample_triangulation` should share code.
I think in an ideal world `track_full_triangulation` should call `track_sample_triangulation` and just create the samples [0, ..., n].
I would prefer minimal code to maintain rather than code duplicates with minor modification.
We can then measure in a second time the speed improvement and memory requirements.

Else it means that the design is bad that we need a ObservationsSampler structure that will use a reference to a whole Observations or create a sub-copy or reference with the right value on the flight if there is a sampling.


Perhaps would be to only have one function `track_obs_triangulation(sfm_data, obs, X)` that we could call also with: `track_obs_triangulation(sfm_data, ObservationSampler(obs, sampler_idx), X)`

Reply:
I agree, I could try to explore the ObservationSampler idea if you want, it's elegant.

Reply:
I'm doing it right now. I will certainly able to push to develop this week end.